include/deal.II/grid/tria_description.h

```diff
@@ -485,6 +527,35 @@ namespace TriangulationDescription
      * triangulation and the TriangulationDescription::Description for all
      * processes in its group, which is communicated.
      *
+     * This function can also be used to read an external mesh only once (by
+     * the root process and a group consisting all processes). The following
+     * code snippet shows a modified version of the example provided in the
+     * documentation of create_description_from_triangulation().
+     * Relevant functions that only need to be performed by the root process
+     * (read and partition mesh) have been into `std::function` objects.
+     *
+     * @code
+     * // create and partition serial triangulation and create description
+     * const TriangulationDescription::Description<dim, spacedim> description =
+     *   TriangulationDescription::Utilities::
+     *     create_description_from_triangulation_in_groups<dim, spacedim>(
+     *       [file_name](auto &tria_base) {
+     *         GridIn<dim, spacedim> grid_in;
+     *         grid_in.attach_triangulation(tria_base);
+     *         grid_in.read(file_name);
+     *       },
+     *       [](auto &tria_base, const auto comm, const auto) {
```

Comment:
can you `/*name*/` that last argument so that it is clear what it represents?

In general, the use of all of these `auto` arguments may make the code shorter, but it makes it not as good as an example because one is left wondering what precisely the functions should be that are passed to `create_description_from_triangulation`.